server/storage/database_test.go

```diff
@@ -440,3 +441,43 @@ func TestDBCheckHealthSucceeds(t *testing.T) {
 	err = dbStore.CheckHealth()
 	require.NoError(t, err)
 }
+
+func TestDBGetChecksum(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	_, store := SetUpSQLite(t, tempBaseDir)
+	defer os.RemoveAll(tempBaseDir)
+
+	ts := data.SignedTimestamp{
+		Signatures: make([]data.Signature, 0),
+		Signed: data.Timestamp{
+			Type:    data.TUFTypes["timestamp"],
+			Version: 0,
+			Expires: data.DefaultExpires("timestamp"),
+		},
+	}
+	j, err := json.Marshal(&ts)
+	require.NoError(t, err)
+	update := MetaUpdate{
+		Role:    data.CanonicalTimestampRole,
+		Version: 1,
+		Data:    j,
+	}
+	checksumBytes := sha256.Sum256(j)
+	checksum := hex.EncodeToString(checksumBytes[:])
+
+	store.UpdateCurrent("gun", update)
+
+	data, err := store.GetChecksum("gun", data.CanonicalTimestampRole, checksum)
+	require.NoError(t, err)
+	require.EqualValues(t, j, data)
+}
+
+func TestDBGetChecksumNotFound(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	_, store := SetUpSQLite(t, tempBaseDir)
+	defer os.RemoveAll(tempBaseDir)
+
+	_, err = store.GetChecksum("gun", data.CanonicalTimestampRole, "12345")
```

Comment:
Could we also add a timestamp to the DB, so that we can be sure that it's the checksum check that's failing?